cms/djangoapps/models/settings/course_metadata.py

```diff
@@ -250,24 +250,43 @@ def validate_and_update_from_json(cls, descriptor, jsondict, user, filter_tabs=T
 
         # Disallow updates to the proctoring provider after course start
         proctoring_provider_model = filtered_dict.get('proctoring_provider')
+
+        # If the user is not edX staff, the user has requested a change to the proctoring_provider
+        # Advanced Setting, and and it is after course start, prevent the user from changing the
+        # proctoring provider.
         if (
             not user.is_staff and
-            proctoring_provider_model != descriptor.proctoring_provider and
+            cls._has_requested_proctoring_provider_changed(descriptor.proctoring_provider, proctoring_provider_model) and
             datetime.now(pytz.UTC) > descriptor.start
         ):
             did_validate = False
             message = (
                 'The proctoring provider cannot be modified after a course has started.'
                 ' Contact {support_email} for assistance'
             ).format(support_email=settings.PARTNER_SUPPORT_EMAIL or 'support')
-            errors.append({'message': message, 'model': filtered_dict.get('proctoring_provider')})
+            errors.append({'message': message, 'model': proctoring_provider_model})
 
         # If did validate, go ahead and update the metadata
         if did_validate:
             updated_data = cls.update_from_dict(key_values, descriptor, user, save=False)
 
         return did_validate, errors, updated_data
 
+    @staticmethod
+    def _has_requested_proctoring_provider_changed(current_provider, requested_provider):
+        """
+        Return whether the requested proctoring provider is different than the current proctoring provider, indicating
+        that the user has requested a change to the proctoring_provider Advanced Setting.
+
+        The requested_provider will be None if the proctoring_provider setting is not available (e.g. if the 
+        ENABLE_PROCTORING_PROVIDER_OVERRIDES waffle flag is not enabled for the course). In this case, we consider
+        that there is no change in the requested proctoring provider.
+        """
+        if current_provider is None or requested_provider is None:
```

Comment:
For those courses with the waffle flag `ENABLE_PROCTORING_PROVIDER_OVERRIDES` turned on, what is the value of the `current_provider` initially? Is it something other than `None`?

Reply:
The value of `current_provider` will normally be the default specified in the `PROCTORING_BACKENDS` Django setting. In our case, that's `software_secure`. But it could be anything. If the `DEFAULT` key is absent from the `PROCTORING_BACKENDS` setting, then `current_provider` will be `None`. 

If the `ENABLE_PROCTORING_PROVIDER_OVERRIDES` waffle flag is enabled, courses still have the `proctoring_provider` setting - it's just hidden from the Advanced Settings page.

There's an edge case where a course could have had the `ENABLE_PROCTORING_PROVIDER_OVERRIDES` waffle flag enabled at one point. Say a user changed the proctoring provider to something other than the default, and then had the waffle flag was disabled. In that case, `current_provider` would be whatever the value of the setting was before the flag was disabled. That's a very unlikely case, I assume, but I wanted to call out that it's not _always_ going to be the default.

Reply:
With your change here, we would have the scenario where, a course had the Proctor_backend set to `None` somehow, before course start. Then when course started, as a course staff, they can set the Proctor_backend to either Software_secure or Proctortrack. This is because their `current_provider` was None. Do we want that?

Reply:
I think that the only case in which a proctoring provider could be set to `None` is if the `DEFAULT` key is missing from `PROCTORING_BACKENDS`. 

It's not actually possible for a user to enter `None` in the Advanced Settings. It gets automatically cast to the string "None", which is then subject to the validation that checks whether "None" is a valid proctoring provider.

Even if someone is able to set the proctoring provider to `None` somehow, which I don't think is possible, [this](https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/course_module.py#L220) would return the default.

So, the only case we have to worry about is if `DEFAULT` is missing. That is an edge case. I think we would still want to prevent a course team member from changing the provider, so I can remove `if current_provider is None` from the condition. I'll think about it more after stand up and reply back.

Reply:
I see. Sounds good then. Thank you for clarify this for me.